Origin:                  https://bugs.gentoo.org/show_bug.cgi?id=426254

diff -ur alsa-plugins-1.0.25.orig/usb_stream/pcm_usb_stream.c alsa-plugins-1.0.25/usb_stream/pcm_usb_stream.c
--- alsa-plugins-1.0.25.orig/usb_stream/pcm_usb_stream.c	2012-01-25 07:56:43.000000000 +0000
+++ alsa-plugins-1.0.25/usb_stream/pcm_usb_stream.c	2012-08-15 01:19:28.131261581 +0100
@@ -18,8 +18,8 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  */
 
-#include <byteswap.h>
 #define _GNU_SOURCE
+#include <byteswap.h>
 #include <sys/mman.h>
 #include <sys/shm.h>
 #include <sys/ioctl.h>
