--- gcc-4.9.2-orig/gcc/config.gcc	2014-09-17 10:16:02.000000000 -0400
+++ gcc-4.9.2/gcc/config.gcc	2015-02-10 22:40:08.000000000 -0500
@@ -1034,6 +1034,13 @@
 	    tmake_file="$tmake_file arm/t-linux-androideabi"
 	    ;;
 	esac
+	case ${target} in
+	arm*-*-*eabihf)
+	    with_cpu=${with_cpu:-cortex-a7}
+	    with_fpu=${with_fpu:-neon-vfpv4}
+	    with_float=${with_float:-hard}
+	    ;;
+	esac
 	# The EABI requires the use of __cxa_atexit.
 	default_use_cxa_atexit=yes
 	with_tls=${with_tls:-gnu}
