Origin:                  http://www.intestinate.com/pilfs

--- gcc-5.2.0-orig/gcc/config.gcc	2015-08-12 22:52:18.307460317 -0400
+++ gcc-5.2.0/gcc/config.gcc	2015-08-12 22:53:18.177531883 -0400
@@ -1050,6 +1050,12 @@
 	    tmake_file="$tmake_file arm/t-linux-androideabi"
 	    ;;
 	esac
+	case ${target} in
+	arm*-*-*eabihf)
+	    with_cpu=${with_cpu:-arm1176jzf-s}
+	    with_float=${with_float:-hard}
+	    ;;
+	esac
 	# The EABI requires the use of __cxa_atexit.
 	default_use_cxa_atexit=yes
 	with_tls=${with_tls:-gnu}
