Origin:                  http://www.intestinate.com/pilfs

--- gcc-4.8.0-orig/gcc/config.gcc	2013-03-31 07:52:42.112209361 -0400
+++ gcc-4.8.0/gcc/config.gcc	2013-03-31 07:57:43.486441714 -0400
@@ -883,6 +883,12 @@
 	    tmake_file="$tmake_file arm/t-linux-androideabi"
 	    ;;
 	esac
+	case ${target} in
+	arm*-*-*eabihf)
+		with_cpu=${with_cpu:-arm1176jzf-s}
+		with_float=${with_float:-hard}
+		;;
+	esac
   	# The BPABI long long divmod functions return a 128-bit value in
 	# registers r0-r3.  Correctly modeling that requires the use of
 	# TImode.
