Origin:                  http://old.nabble.com/-Patch--ARM--Add-arm-linux-gnueabihf-triplet-support.-td33974340.html

diff -ur gcc-4.7.2-orig/gcc/ada/gcc-interface/Makefile.in gcc-4.7.2/gcc/ada/gcc-interface/Makefile.in
--- gcc-4.7.2-orig/gcc/ada/gcc-interface/Makefile.in	2012-08-06 15:34:27.000000000 +0100
+++ gcc-4.7.2/gcc/ada/gcc-interface/Makefile.in	2012-10-04 19:38:42.273277628 +0100
@@ -1867,7 +1867,7 @@
   LIBRARY_VERSION := $(LIB_VERSION)
 endif
 
-ifeq ($(strip $(filter-out arm% linux-gnueabi,$(arch) $(osys)-$(word 4,$(targ)))),)
+ifeq ($(strip $(filter-out arm% linux-gnueabi%,$(arch) $(osys)-$(word 4,$(targ)))),)
   LIBGNAT_TARGET_PAIRS = \
   a-intnam.ads<a-intnam-linux.ads \
   s-inmaop.adb<s-inmaop-posix.adb \
diff -ur gcc-4.7.2-orig/gcc/config.gcc gcc-4.7.2/gcc/config.gcc
--- gcc-4.7.2-orig/gcc/config.gcc	2012-09-12 10:03:54.000000000 +0100
+++ gcc-4.7.2/gcc/config.gcc	2012-10-04 19:44:04.416007071 +0100
@@ -850,7 +850,7 @@
 	esac
 	tmake_file="${tmake_file} arm/t-arm"
 	case ${target} in
-	arm*-*-linux-*eabi)
+	arm*-*-linux-*eabi*)
 	    tm_file="$tm_file arm/bpabi.h arm/linux-eabi.h"
 	    tmake_file="$tmake_file arm/t-arm-elf arm/t-bpabi arm/t-linux-eabi"
 	    # Define multilib configuration for arm-linux-androideabi.
@@ -859,6 +859,15 @@
 		tmake_file="$tmake_file arm/t-linux-androideabi"
 		;;
 	    esac
+	    case ${target} in
+	    arm*-*-*eabihf)
+		with_float=${with_float:-hard}
+		with_fpu=${with_fpu:-vfp}
+		with_arch=${with_arch:-armv6zk}
+		with_cpu=${with_cpu:-arm1176jzf-s}
+		with_tune=${with_tune:-arm1176jzf-s}
+		;;
+	    esac
   	    # The BPABI long long divmod functions return a 128-bit value in
 	    # registers r0-r3.  Correctly modeling that requires the use of
 	    # TImode.
diff -ur gcc-4.7.2-orig/libgcc/config.host gcc-4.7.2/libgcc/config.host
--- gcc-4.7.2-orig/libgcc/config.host	2012-09-05 13:19:47.000000000 +0100
+++ gcc-4.7.2/libgcc/config.host	2012-10-04 19:38:42.303279713 +0100
@@ -327,7 +327,7 @@
 arm*-*-linux*)			# ARM GNU/Linux with ELF
 	tmake_file="${tmake_file} arm/t-arm t-fixedpoint-gnu-prefix"
 	case ${host} in
-	arm*-*-linux-*eabi)
+	arm*-*-linux-*eabi*)
 	  tmake_file="${tmake_file} arm/t-elf arm/t-bpabi arm/t-linux-eabi t-slibgcc-libgcc"
 	  tm_file="$tm_file arm/bpabi-lib.h"
 	  unwind_header=config/arm/unwind-arm.h
diff -ur gcc-4.7.2-orig/libjava/configure gcc-4.7.2/libjava/configure
--- gcc-4.7.2-orig/libjava/configure	2012-09-20 08:23:55.000000000 +0100
+++ gcc-4.7.2/libjava/configure	2012-10-04 19:38:42.383285273 +0100
@@ -20542,7 +20542,7 @@
     # on Darwin -single_module speeds up loading of the dynamic libraries.
     extra_ldflags_libjava=-Wl,-single_module
     ;;
-arm*linux*eabi)
+arm*linux*eabi*)
     # Some of the ARM unwinder code is actually in libstdc++.  We
     # could in principle replicate it in libgcj, but it's better to
     # have a dependency on libstdc++.
diff -ur gcc-4.7.2-orig/libjava/configure.ac gcc-4.7.2/libjava/configure.ac
--- gcc-4.7.2-orig/libjava/configure.ac	2012-05-21 19:14:01.000000000 +0100
+++ gcc-4.7.2/libjava/configure.ac	2012-10-04 19:38:42.393285968 +0100
@@ -931,7 +931,7 @@
     # on Darwin -single_module speeds up loading of the dynamic libraries.
     extra_ldflags_libjava=-Wl,-single_module
     ;;
-arm*linux*eabi)
+arm*linux*eabi*)
     # Some of the ARM unwinder code is actually in libstdc++.  We
     # could in principle replicate it in libgcj, but it's better to
     # have a dependency on libstdc++.
diff -ur gcc-4.7.2-orig/libstdc++-v3/configure.host gcc-4.7.2/libstdc++-v3/configure.host
--- gcc-4.7.2-orig/libstdc++-v3/configure.host	2012-02-10 18:10:12.000000000 +0000
+++ gcc-4.7.2/libstdc++-v3/configure.host	2012-10-04 19:38:42.403286663 +0100
@@ -340,7 +340,7 @@
         fi
     esac
     case "${host}" in
-      arm*-*-linux-*eabi)
+      arm*-*-linux-*eabi*)
 	port_specific_symbol_files="\$(srcdir)/../config/os/gnu-linux/arm-eabi-extra.ver"
 	;;
     esac
