--- glibc-2.22-orig/scripts/test-installation.pl	2015-08-14 19:06:51.543327102 -0400
+++ glibc-2.22/scripts/test-installation.pl	2015-08-14 19:07:51.623396840 -0400
@@ -111,9 +111,11 @@
     # - libthread_db since it contains unresolved references
     # - it's just a test NSS module
     # - We don't provide the libgcc so we don't test it
+    # - libmvec is not implemented on ARM
     if ($name ne "nss_ldap" && $name ne "db1"
 	&& !($name =~/^nss1_/) && $name ne "thread_db"
-	&& $name ne "nss_test1" && $name ne "libgcc_s") {
+	&& $name ne "nss_test1" && $name ne "libgcc_s"
+	&& $name ne "mvec") {
       $link_libs .= " -l$name";
       $versions{$name} = $version;
     }
