Origin:                  http://article.gmane.org/gmane.linux.kernel.modules/1061

--- kmod-13-orig/libkmod/missing.h	2013-05-17 15:55:32.454669764 +0100
+++ kmod-13/libkmod/missing.h	2013-05-17 15:55:47.594064615 +0100
@@ -20,8 +20,15 @@
 #endif
 
 #ifndef HAVE_FINIT_MODULE
+#include <errno.h>
+
 static inline int finit_module(int fd, const char *uargs, int flags)
 {
-	return syscall(__NR_finit_module, fd, uargs, flags);
+	if (__NR_finit_module == -1) {
+		errno = ENOSYS;
+		return -1;
+	} else {
+		return syscall(__NR_finit_module, fd, uargs, flags);
+	}
 }
 #endif
