Origin:                  http://www.intestinate.com/pilfs/#quake3

diff -ur openal-soft-1.14-orig/CMakeLists.txt openal-soft-1.14/CMakeLists.txt
--- openal-soft-1.14-orig/CMakeLists.txt	2012-07-28 19:24:55.980000002 +0100
+++ openal-soft-1.14/CMakeLists.txt	2012-07-28 19:26:30.250000002 +0100
@@ -57,7 +57,7 @@
 
 OPTION(UTILS  "Build and install utility programs"  ON)
 
-OPTION(EXAMPLES  "Build and install example programs"  ON)
+OPTION(EXAMPLES  "Build and install example programs"  OFF)
 
 OPTION(ALSOFT_CONFIG "Install alsoft.conf configuration file" OFF)
 
diff -ur openal-soft-1.14-orig/OpenAL32/Include/alu.h openal-soft-1.14/OpenAL32/Include/alu.h
--- openal-soft-1.14-orig/OpenAL32/Include/alu.h	2012-07-28 19:24:55.960000002 +0100
+++ openal-soft-1.14/OpenAL32/Include/alu.h	2012-07-28 19:26:15.120000002 +0100
@@ -227,14 +227,6 @@
 
 static __inline int SetMixerFPUMode(void)
 {
-#if defined(_FPU_GETCW) && defined(_FPU_SETCW)
-    fpu_control_t fpuState, newState;
-    _FPU_GETCW(fpuState);
-    newState = fpuState&~(_FPU_EXTENDED|_FPU_DOUBLE|_FPU_SINGLE |
-                          _FPU_RC_NEAREST|_FPU_RC_DOWN|_FPU_RC_UP|_FPU_RC_ZERO);
-    newState |= _FPU_SINGLE | _FPU_RC_ZERO;
-    _FPU_SETCW(newState);
-#else
     int fpuState;
 #if defined(HAVE__CONTROLFP)
     fpuState = _controlfp(0, 0);
@@ -243,7 +235,6 @@
     fpuState = fegetround();
     fesetround(FE_TOWARDZERO);
 #endif
-#endif
     return fpuState;
 }
 
