--- udev-lfs-200-1-orig/Makefile.lfs	2013-04-05 12:36:06.621810972 -0400
+++ udev-lfs-200-1/Makefile.lfs	2013-04-05 12:37:08.735034269 -0400
@@ -33,6 +33,7 @@
 
 DEF = -include cfg.h                    \
       -DSYSCONFDIR=\"/etc\"             \
+      -DHAVE_FIRMWARE                   \
       -DFIRMWARE_PATH="\"/lib/firmware/updates/\", \"/lib/firmware/\"" \
       -DHWDB_BIN=\"/etc/udev/hwdb.bin\"                                \
       -DROOTPREFIX=                     \
