Origin:                  http://www.intestinate.com/pilfs/beyond.html#kodi

--- xbmc-14.1-Helix-orig/configure.in	2015-02-16 21:22:07.784946821 +0000
+++ xbmc-14.1-Helix/configure.in	2015-02-16 21:24:50.844604490 +0000
@@ -723,9 +723,9 @@
 case $use_platform in
   raspberry-pi)
      target_platform=target_raspberry_pi
-     use_neon=no
+     use_neon=yes
      use_arch="arm"
-     use_cpu=arm1176jzf-s
+     use_cpu=cortex-a7
      use_hardcoded_tables="yes"
      ARCH="arm"
      AC_DEFINE(HAS_EGLGLES, [1], [Define if supporting EGL based GLES Framebuffer])
--- xbmc-14.1-Helix-orig/tools/depends/configure.in	2015-02-17 02:23:17.984997926 +0000
+++ xbmc-14.1-Helix/tools/depends/configure.in	2015-02-17 02:27:52.514695551 +0000
@@ -289,9 +289,9 @@
   else
     AC_MSG_ERROR([Raspberry Pi firmware not found])
   fi
-  use_neon=no
+  use_neon=yes
   use_arch="arm"
-  use_cpu="arm1176jzf-s"
+  use_cpu="cortex-a7"
   use_hardcoded_tables="yes"
   use_alsa="no"
   ARCH="arm"
@@ -299,13 +299,11 @@
   cross_compiling="yes"
   use_host="arm-linux-gnueabihf"
   deps_dir="$use_host"
-  platform_cflags="-pipe -mcpu=arm1176jzf-s -mtune=arm1176jzf-s -mfloat-abi=hard \
-   -mfpu=vfp -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated \
+  platform_cflags="-pipe -Wno-psabi -Wa,-mno-warn-deprecated \
    -Wno-deprecated-declarations -isystem${use_firmware}/opt/vc/include \
    -isystem${use_firmware}/opt/vc/include/interface/vcos/pthreads \
    -isystem${use_firmware}/opt/vc/include/interface/vmcs_host/linux"
-  platform_cxxflags="-pipe -mcpu=arm1176jzf-s -mtune=arm1176jzf-s \
-   -mfloat-abi=hard -mfpu=vfp -mabi=aapcs-linux -Wno-psabi \
+  platform_cxxflags="-pipe -Wno-psabi \
    -Wa,-mno-warn-deprecated -Wno-deprecated-declarations \
    -isystem${use_firmware}/opt/vc/include \
    -isystem${use_firmware}/opt/vc/include/interface/vcos/pthreads \
